avocado/core/runner.py

```diff
@@ -381,8 +378,20 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
 
         cycle_timeout = 1
         time_started = time.time()
-        proc.start()
 
+        # A fake early status is necessary because to be accurate, the
+        # start_test event should be triggered as closely as possible to
+        # the real start of the test process **AND** it has to be triggered
+        # from the test runner, and not from the test itself
+        test_parameters = test_factory[1]
+        fake_early_status = {'name': test_parameters.get('name'),
+                             'job_logdir': self.job.logdir,
+                             'job_unique_id': self.job.unique_id}
+        self.job._result_events_dispatcher.map_method('start_test',
+                                                      self.job.result,
+                                                      fake_early_status)
+
+        proc.start()
```

Comment:
In general yes, but I'd do this change together with https://trello.com/c/W58vhyHR/539-some-avocado-test-methods-and-attributes-are-public which suggests to move the whole `run_test` into runner. Then the runner would actually be responsible for initializing the test.

Another note, the `start_test` dispatcher accepts `early_status` which should also contain all the info of the initialized test. This is not just `name` and `job_logdir`, but also (test) `logdir` which is suppose to be already created. Therefor I think it should be run from the main process, but after the test's `__init__` is executed (but not the `run_avocado`) as it might be important to plugins.